File: C++/Cristian Maldonado/Jump_Search.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

int jumpSearch(int arr[], int x, int tam){
	int step = (int) floor(sqrt(tam));
	int prev = 0;
	while(arr[min(step,tam)-1] < x){
		prev = step;
		step += (int) floor(sqrt(tam));
		if(prev >= tam){
			return -1;
		}
	}
	while(arr[prev] < x){
		prev++;
		if(prev == min (step,tam)){
			return -1;
		}
	}
	if(arr[prev] == x){
		return prev;
	}
	return -1;
}

int main (int argc, char *argv[]) {
	int arr[] = {0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610};
	int x = 144;
	int resultado = jumpSearch(arr, x, sizeof(arr)/sizeof(0));
	if(resultado == -1){
		cout << "No encontrado" << endl;
	}else{
		cout << "Encontrado en la posicion: " + std::to_string(resultado) << endl;
	}
	return 0;
}

File: PY/Cristian Maldonado/Jump_Search.py
import math

def jumpSearch(arr , x, tam):
    step = int(math.floor(math.sqrt(tam)))
    prev = 0
    while(arr[min(step, tam)-1] < x):
        prev = step
        step = step + int(math.floor(math.sqrt(tam)))
        if(prev >= tam):
            return -1
    while(arr[prev] < x):
        prev = prev + 1
        if(prev == min(step ,tam)):
            return -1
    if(arr[prev] == x):
        return prev
    return -1

arr = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610]
buscado = 610
resultado = jumpSearch(arr, buscado,len(arr))
if(resultado == -1):
    print("Elemento no presente")
else:
    print("Elemento encontrado en la posicion",resultado)
